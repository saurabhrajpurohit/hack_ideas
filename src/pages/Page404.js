import { Link as RouterLink } from 'react-router-dom';
import { styled } from '@mui/material/styles';
import { Box, Button, Typography, Container } from '@mui/material';

const RootStyle = styled('div')(({ theme }) => ({
    display: 'flex',
    minHeight: '100%',
    alignItems: 'center',
    paddingTop: theme.spacing(15),
    paddingBottom: theme.spacing(10)
}));

export default function Page404() {
    return (
        <RootStyle data-testid="page-404">
            <Container>
                <Box sx={{ maxWidth: 480, margin: 'auto', textAlign: 'center' }}>
                    <Typography variant="h3" paragraph>
                        Sorry, page not found!
                    </Typography>
                    <Typography sx={{ color: 'text.secondary' }}>
                        Sorry, we couldn’t find the page you’re looking for. Perhaps you’ve mistyped the URL?
                        Be sure to check your spelling.
                    </Typography>

                    <Button sx={{ mt: 3 }} to="/" size="large" variant="contained" component={RouterLink}>
                        Go to Home
                    </Button>
                </Box>
            </Container>
        </RootStyle>
    );
}
