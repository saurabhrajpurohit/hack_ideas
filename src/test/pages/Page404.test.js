import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Page404 from '../../pages/Page404';

test('ui to be rendered correctly', () => {
    render(
        <BrowserRouter>
            <Page404 />
        </BrowserRouter>
    );
    const element = screen.getByTestId('page-404');
    expect(element).toMatchSnapshot();
});
