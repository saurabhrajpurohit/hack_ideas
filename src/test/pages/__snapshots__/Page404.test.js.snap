// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui to be rendered correctly 1`] = `
<div
  class="css-10jlxvb"
  data-testid="page-404"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-95xhe1"
    >
      <p
        class="MuiTypography-root MuiTypography-h3 MuiTypography-paragraph css-1ubdccs-MuiTypography-root"
      >
        Sorry, page not found!
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
      >
        Sorry, we couldn’t find the page you’re looking for. Perhaps you’ve mistyped the URL? Be sure to check your spelling.
      </p>
      <a
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-fyrgu4-MuiButtonBase-root-MuiButton-root"
        href="/"
        tabindex="0"
      >
        Go to Home
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
